### PR TITLE
chore: Initialize dev-upgrade-17

### PR DESCRIPTION
**Pull request by gibson042**
## Description
Bumps the cosmos upgrade handler and renames a3p-integration/proposals/n:upgrade-next to a:upgrade-17. This is mostly as described in [MAINTAINERS.md](https://github.com/Agoric/agoric-sdk/blob/master/MAINTAINERS.md#create-the-dev-release-branch), but the existing [a3p-integration/package.json `"fromTag": "use-vaults-auctions"`](https://github.com/Agoric/agoric-sdk/blob/e67cd918f33dafb1708bd4341f2a737b79fa574f/a3p-integration/package.json#L4) still seems correct and leaving "s:stake-bld" and "z:acceptance" proposals rather than merging them into "a:upgrade-17".

### Security Considerations
n/a

### Scaling Considerations
n/a

### Documentation Considerations
To be included in the full release.

### Testing Considerations
No post-proposal tests are added, but they may come in later pull requests.

### Upgrade Considerations
This will go through the usual validation.

### CODE REVIEW

File: golang/cosmos/app/upgrade.go
Comment (mhofman) [inline golang/cosmos/app/upgrade.go]: Ah I now remember another change we decided in the last upgrade: drop the `agorictest` upgrade handler names, and instead make sure we have a `agoric-upgrade-17-2` available to re-apply the upgrade if necessary.

I suppose this may require a change in the release notes script as well.

https://github.com/Agoric/agoric-sdk/blob/agoric-upgrade-16/golang/cosmos/app/upgrade.go#L14-L15

File: golang/cosmos/app/upgrade.go
Comment (mhofman) [inline golang/cosmos/app/upgrade.go:1]: There is a change missing to `isPrimaryUpgradeName`

**Review by mhofman — CHANGES_REQUESTED** [review]
There is a change missing to `upgrade.go` regarding `isPrimaryUpgradeName`.

> the existing [a3p-integration/package.json `"fromTag": "use-vaults-auctions"`](https://github.com/Agoric/agoric-sdk/blob/e67cd918f33dafb1708bd4341f2a737b79fa574f/a3p-integration/package.json#L4) still seems correct

Right, this got changed in https://github.com/Agoric/agoric-sdk/pull/10056 as we decided to move away from always using latest.

> leaving "s:stake-bld" and "z:acceptance" proposals rather than merging them into "a:upgrade-17"

We may have decided to introduce acceptance testing as a "proposal" after these instructions was written.

I'll open a separate PR to update the maintainers and possibly the release script. Issue https://github.com/Agoric/agoric-sdk/issues/10089

File: golang/cosmos/app/upgrade.go
Comment (gibson042) [reply golang/cosmos/app/upgrade.go:1]: Yeah, thanks. I think I'd like to refactor that function, but will do so on master. Updated here to keep the current form.

Comment (gibson042) [reply golang/cosmos/app/upgrade.go:1]: #10091

**Review by mhofman — APPROVED** [review]
I'll leave it to you whether to merge this now, or after #10091 lands on master (in which case I'll push `dev-upgrade-17` forward

**Review by LuqiPan — APPROVED** [review]
LGTM

**Pull request merged**
